src/Application/GamePathResolver.cpp

```diff
@@ -0,0 +1,79 @@
+﻿#include <vector>
+
+#include "Application/GamePathResolver.h"
+
+#include "Engine/IocContainer.h"
+#include "Library/Logger/Logger.h"
+
+
+using namespace Application;
+using EngineIoc = Engine_::IocContainer;
+
+
+static std::string _ResovlePath(Platform *platform, const char *envVarOverride, const std::vector<const wchar_t*> &registryKeys);
```

Comment:
Typo in Resolve.

Also, why forward-declare?

Reply:
i wanted the important/public/interface functions on top (those are exported in .h) and then the impl details on the bottom.
thought that would contribute to the readability
opinions?

Reply:
I'm fine with either approach. Personally, I usually place the static functions on top.